File: finalProject/finalProjectFrontend/parkplay_frontend/lib/models/parking_lot.dart
import 'parking_spot.dart';

class ParkingLot {
  final int id;
  final String name;
  final String location;
  final int? managerId;
  final List<ParkingSpot> spots;

  ParkingLot({
    required this.id, 
    required this.name, 
    required this.location, 
    this.managerId, 
    required this.spots
  });

  factory ParkingLot.fromJson(Map<String, dynamic> json) {
    var spotsList = json['spots'] as List?;
    List<ParkingSpot> parsedSpots = spotsList != null 
        ? spotsList.map((i) => ParkingSpot.fromJson(i)).toList() 
        : [];
    
    return ParkingLot(
      id: json['id'],
      name: json['name'],
      location: json['location'],
      managerId: json['managerId'],
      spots: parsedSpots,
    );
  }
}

File: finalProject/finalProjectBackend/ParkplayFinal/src/main/java/org/example/parkplayfinal/model/ParkingLot.java
package org.example.parkplayfinal.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

/**Reprezinta o parcare care contine locuri de parcare, are id, nume si locatie si este asociata unui user cu rolul de parking manager*/
@Data
@NoArgsConstructor
public class ParkingLot {
    private Long id;
    private String name;
    private String location;
    private Long managerId;
    private List<ParkingSpot> spots = new ArrayList<>();

    public ParkingLot(String name, String location, Long managerId) {
        this.name = name;
        this.location = location;
        this.managerId = managerId;
    }

    public void addSpot(ParkingSpot spot) {
        this.spots.add(spot);
    }
}